### PR TITLE
Blockatlas - Discard and retry BTC blocks with 0 transactions

### PR DESCRIPTION
**Pull request by akinmail**

### CODE REVIEW

**Review by logbon72 — DISMISSED** [review]

File: observer/stream.go [hunk 1]
```diff
@@ -100,7 +101,20 @@ func (s *Stream) loadBlock(c chan<- *blockatlas.Block, num int64) {
...
+			logger.Error("no of transaction in the BTC block is 0, so skipping and retrying again", s.logParams,
+				logger.Params{"block": num, "coin": s.coin})
+		}
+	} else {
+		s.updateBlockHeight(newNum, num)
+	}
+}
+
+func (s *Stream) updateBlockHeight(newNum int64, num int64) {
+	err := s.Tracker.SetBlockNumber(s.coin, newNum)
```
Comment (eddywm) [inline observer/stream.go]:
```suggestion
		if s.coin == coin.BTC && len(block.Txs) < 1 {
		logger.Error("Got an empty BTC block - Not persisting block height number", s.logParams, logger.Params{"block": num, "coin": s.coin})
		return
	  }
```

Comment (eddywm) [reply observer/stream.go]: It looks like too much.

We can simply check if it's BTC and length < 1 and just return right away,
Only update block height if it meets criteria.

File: observer/stream.go [hunk 2]
```diff
@@ -100,7 +101,10 @@ func (s *Stream) loadBlock(c chan<- *blockatlas.Block, num int64) {
 	// Not strictly correct nor avoids race conditions
 	// But good enough
 	newNum := atomic.AddInt64(&s.blockNumber, 1)
-	err = s.Tracker.SetBlockNumber(s.coin, newNum)
```
Comment (logbon72) [inline observer/stream.go:103]: What happened here? It's removed forever.

Comment (logbon72) [reply observer/stream.go:103]: We need to restore this line after the if-block

Comment (eddywm) [inline observer/stream.go:103]: Hmm  looks like you have now removed the actual persisting of block height though.

The idea is to check and return right away.

And then for those meeting, they should persisted in the end.

File: observer/stream.go [hunk 3]
```diff
@@ -100,7 +101,10 @@ func (s *Stream) loadBlock(c chan<- *blockatlas.Block, num int64) {
...
+		if s.coin == coin.BTC && len(block.Txs) < 1 {
```
Comment (eddywm) [inline observer/stream.go]: Please fix indentation.

File: observer/stream.go (hunk 2 above)
Comment (akinmail) [reply observer/stream.go:103]: I only commited Eddys suggestion. Will review

Comment (akinmail) [reply observer/stream.go:103]: I only commited your suggestion. Will review

File: observer/stream.go (hunk 3 above)
Comment (akinmail) [reply observer/stream.go]: Done

File: observer/stream.go (hunk 2 above)
Comment (akinmail) [reply observer/stream.go:103]: Done

Comment (akinmail) [reply observer/stream.go:103]: Done

Comment (eddywm) [reply observer/stream.go:103]: @akinmail you haven't pushed it seems

**Review by logbon72 — APPROVED** [review]

**Review by eddywm — APPROVED** [review]

**Pull request merged**